projects/step-core/src/lib/shared/utils.ts

```diff
@@ -107,11 +108,60 @@ export const toRecord = <T>(keyValuePairs: KeyValue<string, T>[]): Record<string
       ...acc,
       [key]: value,
     }),
-    {}
+    {},
   );
 
 export const routesPrioritySortPredicate = (routeA: Route, routeB: Route) => {
   const weightA = routeA.data?.[SUB_ROUTE_DATA]?.weight ?? 1;
   const weightB = routeB.data?.[SUB_ROUTE_DATA]?.weight ?? 1;
   return weightA - weightB;
 };
+
+export const getIconAndPageById = (id: string): Bookmark | undefined => {
```

Comment:
As this adds a large function solely for some meta info of bookmarks I would remove it.